packages/apollo/src/utils/lists.ts

```diff
@@ -21,6 +21,45 @@ export type ListHelperOptions = {
  */
 export type ClientPaginationBehaviour = 'paginate' | 'showAll'
 
+/**
+ * Checks if the provided value is an Apollo Client Reference object
+ * Used to determine if a value in the cache is a reference to another cached object
+ * 
+ * @param link - Value to check for Reference type
+ * @returns True if the value is an Apollo Reference object, false otherwise
+ */
+function hasRef (link: unknown): link is Reference {
+    return (
+        typeof link === 'object' &&
+        link !== null &&
+        '__ref' in link &&
+        typeof link.__ref === 'string'
+    )
+
+}
+
+/**
+ * Validates if a list of cached items contains any broken references
+ * A broken reference occurs when a Reference object points to a cache entry that doesn't exist
+ * This is used to ensure data integrity when reading from the Apollo cache
+ * 
+ * @param list - Array of potentially cached items to check
+ * @param cache - Apollo normalized cache object
+ * @returns True if the list contains any broken references or non-reference items, false if all references are valid
```

Comment:
Why this is default behaviour?

Reply:
If I have link `and` its not in cache (expired) => we need to refetch
If I have no link (`null` or some other values), then its is a valid list, isn't it?